core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/ApplyTableScanRedirection.java

```diff
@@ -92,7 +93,10 @@ public Result apply(TableScanNode scanNode, Captures captures, Context context)
         if (destinationTableHandle.isEmpty()) {
             throw new TrinoException(TABLE_NOT_FOUND, format("Destination table %s from table scan redirection not found", destinationTable));
         }
-        if (destinationTableHandle.get().equals(scanNode.getTable())) {
+
+        TableMetadata tableMetadata = metadata.getTableMetadata(context.getSession(), scanNode.getTable());
+        CatalogSchemaTableName sourceTableName = new CatalogSchemaTableName(tableMetadata.getCatalogName().getCatalogName(), tableMetadata.getTable());
+        if (destinationTable.equals(sourceTableName)) {
```

Comment:
Add a comment why this change is needed. Could we have a test?

Reply:
This becomes necessary because the table handle from scan node could contain projected column handles but the destination table handle won't have them. So the redirection could have already taken place but we would fail to detect it due to the difference in table handle.
One of the existing redirection tests failed after I updated MockConnectorTableHandle#equals method